File: Sources/Chicane/Core/Math/Vec/Vec2.cpp
#include "Chicane/Core/Math/Vec/Vec2.hpp"

#include "Chicane/Core/Math/Vec/Vec3.hpp"

namespace Chicane
{
    const Vec2 Vec2::Zero = Vec2(0.0f);
    const Vec2 Vec2::One  = Vec2(1.0f);

    const Vec2 Vec2::Up    = Vec2(Vec3::Up.x,    Vec3::Up.y);
    const Vec2 Vec2::Right = Vec2(Vec3::Right.x, Vec3::Right.y);
}

File: Includes/Chicane/Core/Script/Types/Vec2.lua
---@meta

---@class Vec2
---@field x number
---@field y number
Vec2 = {}

---@param x number
---@param y number
---@return Vec2
function Vec2(x, y) end

---@return Vec2
function Vec2.Zero() end

---@return Vec2
function Vec2.One() end

---@return Vec2
function Vec2.Right() end

---@return Vec2
function Vec2.Up() end

---@return string
function Vec2:toString() end
